src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/HtmlDoclet.java

```diff
@@ -304,31 +301,25 @@ protected void generateOtherFiles(ClassTree classTree)
 
         // If a stylesheet file is not specified, copy the default stylesheet
         // and replace newline with platform-specific newline.
-        DocFile f;
         if (options.stylesheetFile().length() == 0) {
-            f = DocFile.createFileForOutput(configuration, DocPaths.STYLESHEET);
-            f.copyResource(DocPaths.RESOURCES.resolve(DocPaths.STYLESHEET), true, true);
+            copyResource(DocPaths.STYLESHEET, DocPaths.RESOURCE_FILES.resolve(DocPaths.STYLESHEET), true);
         }
-        f = DocFile.createFileForOutput(configuration, DocPaths.JAVASCRIPT);
-        f.copyResource(DocPaths.RESOURCES.resolve(DocPaths.JAVASCRIPT), true, true);
-        f = DocFile.createFileForOutput(configuration, DocPaths.CLIPBOARD_SVG);
-        f.copyResource(DocPaths.RESOURCES.resolve(DocPaths.CLIPBOARD_SVG), true, true);
-        f = DocFile.createFileForOutput(configuration, DocPaths.LINK_SVG);
-        f.copyResource(DocPaths.RESOURCES.resolve(DocPaths.LINK_SVG), true, true);
-        if (options.createIndex()) {
-            f = DocFile.createFileForOutput(configuration, DocPaths.SEARCH_JS);
-            f.copyResource(DOCLET_RESOURCES.resolve(DocPaths.SEARCH_JS_TEMPLATE), configuration.docResources);
-
-            f = DocFile.createFileForOutput(configuration, DocPaths.SEARCH_PAGE_JS);
-            f.copyResource(DOCLET_RESOURCES.resolve(DocPaths.SEARCH_PAGE_JS), configuration.docResources);
+        copyResource(DocPaths.SCRIPT_JS, DocPaths.SCRIPT_FILES.resolve(DocPaths.SCRIPT_JS), true);
```

Comment:
In general, in this method, there is a lot of code of the form `DocPaths.SOME_DIR.resolve(DocPaths.SOME_FILE)`.

In general, I think it is a side-effect of prior "bad" design that these files used to be in the root directory of the generated API, resulting in it being sufficient to define just a simple filename in `DocPaths`.

In general, I think it would be better to move these `resolve` calls into `DocPaths`, so that it becomes a single reference (i.e. without any `.resolve`) here in this code.

As an example, I grepped for SCRIPT_JS:
```
open/src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/HtmlDoclet.java:        copyResource(DocPaths.SCRIPT_JS, DocPaths.SCRIPT_FILES.resolve(DocPaths.SCRIPT_JS), true);
open/src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/markup/Head.java:            addScriptElement(head, DocPaths.SCRIPT_FILES.resolve(DocPaths.SCRIPT_JS));
open/src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/DocPaths.java:    public static final DocPath SCRIPT_JS = DocPath.create("script.js");
```

In both use sites, the usage is: `DocPaths.SCRIPT_FILES.resolve(DocPaths.SCRIPT_JS)` suggesting that the declaration should really be something like:
```
open/src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/DocPaths.java:    public static final DocPath SCRIPT_JS = SCRIPT_FILES.resolve("script.js");
```